great_ape_safe/ape_api/curve.py

```diff
@@ -66,36 +76,31 @@ def _get_coin_indices(self, lp_token, pool, asset_in, asset_out):
                     j = coins.index(coin)
             assert i != None and j != None
         else:
-            registry = self._get_registry(lp_token)
-            i, j, _ = registry.get_coin_indices(
-                pool, asset_in, asset_out
-            )
+            registry = self._get_registry(pool)
+            # workaround; crypo_registry only returns 2 vars, whereas old style
+            # returned 3
+            indices = registry.get_coin_indices(pool, asset_in, asset_out)
+            i = indices[0]
+            j = indices[1]
         return i, j
 
 
     def _get_n_coins(self, lp_token):
-        registry = self._get_registry(lp_token)
         pool = self._get_pool_from_lp_token(lp_token)
-        if not self.is_v2 and registry == self.metapool_registry:
+        registry = self._get_registry(pool)
+        if not registry == self.registry:
             return registry.get_n_coins(pool)
-        else:
-            return registry.get_n_coins(pool)[0] # note [1] tells us if there is a wrapped coin!
+        return registry.get_n_coins(pool)[0]
+
+
+    def _pool_has_wrapped_coins(self, pool):
+        registry = self._get_registry(pool)
+        try:
+            registry.get_underlying_balances(pool)
+            return True
+        except VirtualMachineError:
```

Comment:
for me locally here seems to jump instead a `AttributeError`, do you get other type of exception?